weave/flow/scorers/hallucination_scorer.py

```diff
@@ -0,0 +1,160 @@
+
+# Steps
+1. Carefully read and understand the input data.
+2. Examine the model output.
+3. Compare the output to the input data, identifying any inconsistencies or additions.
+4. Evaluate the logical connection between input and output.
+5. Determine if any information in the output is not supported by or conflicts with the input.
+
+# Guidelines
+- Focus on factual accuracy and logical consistency
+- Consider both explicit and implicit information in the input data
+- Be aware of potential misinterpretations or over-generalizations in the output
+- Identify any information in the output that goes beyond the scope of the input
+
+# Examples
+## Data to analyze
+
+<input_data_example>
+The cat is black and white.
+</input_data_example>
+
+<output_example>
+The cat has orange stripes.
+</output_example>
+
+## Analysis:
+{
+  "think_step_by_step": "The cat is black and white. The cat has orange stripes. \
+The output contradicts the input data because the input specifies black and white, \
+while the output mentions orange. The output also introduces a pattern not present in \
+the input.",
+  "reasoning": [
+    {
+      "hallucination_type": "Color comparison",
+      "observation": "Input specifies black and white, output mentions orange"
+    },
+    {
+      "hallucination_type": "Pattern analysis",
+      "observation": "Input doesn't mention any pattern, output introduces stripes"
+    }
+  ],
+  "conclusion": "The output contains two hallucinations: it contradicts the color information \
+and introduces a pattern not present in the input."
+  "is_hallucination": true,
+}
+
+# Notes
+- Ensure each step in the reasoning process is clearly articulated
+- Be objective and avoid assumptions not supported by the input data
+- If the output contains factual information not present in the input, it may be a \
+hallucination even if it doesn't directly contradict the input
+"""
+
+DEFAULT_HALLUCINATION_USER_PROMPT = """
+Analyze the following <input_data> and <output> and determine if the <output> contains any hallucinations.
+# Data to analyze
+
+<input_data>
+{input_data}
+</input_data>
+
+<output>
+{output}
+</output>
+"""
+
+
+class HallucinationReasoning(BaseModel):
+    hallucination_type: str = Field(
+        description="A short name for the type of hallucination."
+    )
+    observation: str = Field(
+        description="An observation from the <input_data> and <output> that supports the hallucination."
+    )
+
+
+class HallucinationResponse(BaseModel):
+    chain_of_thought: str = Field(
+        description="Think step by step about whether the <output> contains hallucinations \
+based on the <input_data>."
+    )
+    reasonings: List[HallucinationReasoning] = Field(
+        description="A list of reasoning steps that lead to the conclusion about whether or not\
+the <output> contains hallucinations."
+    )
+    conclusion: str = Field(description="The conclusion of the analysis.")
+    hallucination_free: bool = Field(
```

Comment:
nit: `True` should generally be the positive case, so instead of `hallucination_free=True` it should probably be `has_hallucination=True`